File: code_samples/kotlin/data_structures_hash_table.kt
/**
 * Data Structures: Hash Table
 * AI/ML Training Sample
 */

class HashTable {
    var data: String = ""
        private set
    
    fun process(input: String) {
        data = input
    }
    
    fun getData(): String = data
    
    fun validate(): Boolean = data.isNotEmpty()
}

fun main() {
    val instance = HashTable()
    instance.process("example")
    println("Data: ${instance.getData()}")
    println("Valid: ${instance.validate()}")
}

File: code_samples/dart/data_structures_hash_table.dart
// Data Structures: Hash Table
// AI/ML Training Sample

class HashTable {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = HashTable();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}
